server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/OperatorTreeGenerator.java

```diff
@@ -639,6 +645,24 @@ public Operator visitNodePathsCount(NodePathsCountNode node, LocalExecutionPlanC
     return new NodePathsCountOperator(operatorContext, child);
   }
 
+  @Override
+  public Operator visitSingleDeviceView(
+      SingleDeviceViewNode node, LocalExecutionPlanContext context) {
+    OperatorContext operatorContext =
+        context
+            .getInstanceContext()
+            .addOperatorContext(
+                context.getNextOperatorId(),
+                node.getPlanNodeId(),
+                SingleDeviceViewOperator.class.getSimpleName());
+    Operator child = node.getChild().accept(this, context);
+    List<Integer> deviceColumnIndex = node.getDeviceToMeasurementIndexes();
+    List<TSDataType> outputColumnTypes = getOutputColumnTypes(node, context.getTypeProvider());
```

Comment:
This will cause each `SingleDeviceView` create a new `List<TsDataType>`, you should put `List<TsDataType>` in `LocalExecutionPlanContext` while visiting MergeSortNode and then reuse that in each `SingleDeviceView`